resources/js/components/YourBalancePage.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <v-row class="pa-0 ma-0">
+    <v-col class="ma-0 pa-0 secondary" cols="12">
+      <the-balance-header :balance="balance" />
+    </v-col>
+    <v-col class="py-0" cols="12">
+      <v-alert
+        v-if="isUploading"
+      >
+        <v-progress-circular
+          indeterminate
+          width="2"
+        />
+        We're uploading {{ uploadCount }} balance entries. Sit tight.
+      </v-alert>
+      <the-entries-list :entries="visibleEntries" />
+    </v-col>
+  </v-row>
+</template>
+
+<script>
+import TheEntriesList from '@/js/components/TheEntriesList'
+import TheBalanceHeader from '@/js/components/TheBalanceHeader'
+  export default {
+    name: 'YourBalancePage',
+    components: {
+      TheEntriesList,
+      TheBalanceHeader,
+    },
+    data () {
+      return {
+        entriesPerPage: 100,
+        isUploading: false,
+        page: 1,
+        user: this.$store.state.user,
+      }
+    },
+    computed: {
+      balance () {
+        return this.entries.reduce((bal, e) => bal + parseFloat(e.value), 0)
```

Comment:
Do you see any issues with calculating the user's total balance on the front-end like this?

Reply:
As noted above, my approach was essentially to fetch ALL of a user's entries at page load. If it's possible that entries are updated on the server **after** they've been fetched then the balance on the client might get out of sync, but realistically, unless you're constantly polling the server, set up pub/sub, or have a websocket connection you're always going to run the risk of the balance getting out of sync regardless of where it's actually calculated.

The balance is a computed property and updates reactively whenever an entry is added, changed, or removed, so short of the data on the client getting out of sync with the server, it should always be accurate.

I suppose one might worry about performance on the client if there were a very large number of entries. The balance is calculated in a top-level component, so, given the current interface, it's unlikely that a re-render is going to force a re-calculation. The calculation could rather trivially be moved to the store, as well.